python/rmm/_cuda/gpu.py

```diff
@@ -126,16 +95,13 @@ def getDeviceCount():
     This function automatically raises CUDARuntimeError with error message
     and status code.
     """
-
-    cdef int count
-    cdef cudaError_t status = cudaGetDeviceCount(&count)
-
-    if status != cudaError.cudaSuccess:
+    status, count = cudart.cudaGetDeviceCount()
+    if status != cudart.cudaError_t.cudaSuccess:
         raise CUDARuntimeError(status)
     return count
 
 
-def getDeviceAttribute(cudaDeviceAttr attr, int device):
+def getDeviceAttribute(attr: cudart.cudaDeviceAttr, device: int):
```

Comment:
Can you explain why we don't get rid of these functions altogether if they exist now in the `cudart` Python module?

Reply:
I think this goes back to Ashwin's comment in the OP. Namely that there are some things that Python users expect that are currently not handled in the low-level bindings that CUDA Python provides. There has been discussion about having a higher level API in CUDA Python, which would handle these needs at which point this code could be dropped.

Taking this particular function as an example. Most Python users would expect an exception to be raised if something doesn't work. Currently CUDA Python returns an error code, which Python users wouldn't necessarily expect or know how to handle.

Reply:
Both cuDF and cuGraph rely on these high-level wrappers around the driver/runtime APIs. Eventually, we expect CUDA  Python to itself provide similar high-level wrappers, but for now, I decided to keep these to avoid the duplication of effort downstream.